UserService/Repository/UserRepository.cs

```diff
@@ -12,41 +13,57 @@ public UserRepository(DataContext dbContext)
     {
         _dbContext = dbContext;
     }
-    public bool CreateUser(User user)
+
+    public void CreateUser(User user)
     {
         _dbContext.Add(user);
-        return Save();
+        Save();
     }
 
     public IEnumerable<User> GetAllUsers()
     {
-        return _dbContext.Users.AsEnumerable();
+        var users = _dbContext.Users.AsEnumerable();
+        return users;
     }
 
     public User GetUser(string id)
     {
-        return _dbContext.Users.Where(u => u.Id == id).FirstOrDefault();
+        var user = _dbContext.Users.Where(u => u.Id == id).FirstOrDefault();
+
+        if (user == null)
```

Comment:
I prefer that repositories only handle what they are meant to do without managing edge cases.

Reply:
I agree, maybe we should add an extra service layer between the repo and the controller to handle this